File: src/main/java/com/example/shopease/payment/infrastructure/entities/Payment.java
package com.example.shopease.payment.infrastructure.entities;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "payments")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Payment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "transaction_id", unique = true, nullable = false)
    private String transactionId;
    
    @Column(name = "order_id", nullable = false)
    private Long orderId;
    
    @Column(name = "amount", nullable = false, precision = 10, scale = 2)
    private BigDecimal amount;
    
    @Column(name = "currency", length = 3)
    @Builder.Default
    private String currency = "BDT";
    
    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private PaymentStatus status;
    
    @Column(name = "gateway_type")
    @Builder.Default
    private String gatewayType = "SSLCOMMERZ";
    
    @Column(name = "ssl_session_id")
    private String sslSessionId;
    
    @Column(name = "gateway_transaction_id")
    private String gatewayTransactionId;
    
    @Column(name = "bank_transaction_id")
    private String bankTransactionId;
    
    @Column(name = "card_type")
    private String cardType;
    
    @Column(name = "card_brand")
    private String cardBrand;
    
    @Lob
    @Column(name = "gateway_response")
    private String gatewayResponse;
    
    @Column(name = "failure_reason")
    private String failureReason;
    
    @Column(name = "customer_name")
    private String customerName;
    
    @Column(name = "customer_email")
    private String customerEmail;
    
    @Column(name = "customer_phone")
    private String customerPhone;
    
    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
    
    public enum PaymentStatus {
        PENDING,
        PROCESSING,
        SUCCESS,
        FAILED,
        CANCELLED,
        REFUNDED,
        PARTIALLY_REFUNDED
    }
    
    public enum PaymentGateway {
        SSLCOMMERZ("SSLCommerz"),
        STRIPE("Stripe"),
        PAYPAL("PayPal");
        
        private final String displayName;
        
        PaymentGateway(String displayName) {
            this.displayName = displayName;
        }
        
        public String getDisplayName() {
            return displayName;
        }
    }
}

File: client/types/payment.ts
// Payment related types
export interface Payment {
  id: string;
  orderId: string;
  transactionId: string;
  amount: number;
  currency: string;
  status: PaymentStatus;
  paymentMethod: string;
  gatewayResponse?: any;
  createdAt: string;
  updatedAt: string;
}

export interface InitiatePaymentRequest {
  orderId: string;
  amount: number;
  currency?: string;
  paymentMethod?: string;
}

export interface InitiatePaymentResponse {
  paymentUrl: string;
  transactionId: string;
  sessionKey?: string;
}

export interface PaymentStatusResponse {
  transactionId: string;
  status: PaymentStatus;
  amount: number;
  currency: string;
  orderId: string;
  paymentMethod: string;
  gatewayResponse?: any;
}

export interface PaymentCallbackRequest {
  transactionId: string;
  status: string;
  amount?: number;
  orderId?: string;
  gatewayResponse?: any;
}

export interface PaymentTestConfig {
  testMode: boolean;
  storeId: string;
  storePassword: string;
  gatewayUrl: string;
}

export enum PaymentStatus {
  PENDING = 'PENDING',
  PROCESSING = 'PROCESSING',
  COMPLETED = 'COMPLETED',
  FAILED = 'FAILED',
  CANCELLED = 'CANCELLED',
  REFUNDED = 'REFUNDED'
}